packages/react-components/react-avatar/src/components/AvatarGroup/useAvatarGroup.tsx

```diff
@@ -0,0 +1,117 @@
+import * as React from 'react';
+import { Button } from '@fluentui/react-button';
+import { extraAvatarGroupClassNames } from './useAvatarGroupStyles';
+import { getInitials } from '../../utils/getInitials';
+import { getNativeElementProps, resolveShorthand } from '@fluentui/react-utilities';
+import { Label } from '@fluentui/react-label';
+import { MoreHorizontalRegular } from '@fluentui/react-icons';
+import { PopoverSurface } from '@fluentui/react-popover';
+import { useId } from '@fluentui/react-utilities';
+import type { AvatarGroupProps, AvatarGroupState } from './AvatarGroup.types';
+
+/**
+ * Create the state required to render AvatarGroup.
+ *
+ * The returned state can be modified with hooks such as useAvatarGroupStyles_unstable,
+ * before being passed to renderAvatarGroup_unstable.
+ *
+ * @param props - props from this instance of AvatarGroup
+ * @param ref - reference to root HTMLElement of AvatarGroup
+ */
+export const useAvatarGroup_unstable = (props: AvatarGroupProps, ref: React.Ref<HTMLElement>): AvatarGroupState => {
+  const {
+    children,
+    layout = 'spread',
+    maxAvatars = 5,
+    overflowIndicator = 'number-overflowed',
+    size = 32,
+    ...rest
+  } = props;
+  const id = useId('avatarGroup-', props.id);
+  const childrenCount = React.Children.count(children);
+  const numOfAvatarsToShow = layout === 'pie' ? 3 : maxAvatars;
+  const hasOverflow = childrenCount > numOfAvatarsToShow;
+  const childrenArray = React.Children.toArray(children);
+
+  const rootChildren = childrenArray.slice(0, numOfAvatarsToShow).map(child => {
+    if (React.isValidElement(child)) {
+      // If the layout is pie and the size is less than 40, the Avatar should only display the first initial
+      let initials = child.props.initials;
+
+      if (layout === 'pie' && child.props.name && size < 40) {
+        // No need to check if it's rtl since we will use only the first letter.
+        initials = getInitials(child.props.name, false)[0];
+      }
```

Comment:
Messing around with the props of Avatar like this is a little questionable... Avatar has its own logic for handling initials and this can lead to buggy and hard to debug behavior of Avatar.

It might be best to instead have AvatarGroup create a context that lets the avatars inside change their behavior based on the group's props.

For example, if Avatar knows its inside a group with pie layout, then it would only render the first initial. And if it's inside a group at all, it would change its default value of `color` to `'colorful'`.

 See RadioGroup and Radio for an example to follow.